src/activities/associateGrade/GradeSchemeLinkedEntity.js

```diff
@@ -28,12 +29,27 @@ export class GradeSchemeLinkedEntity extends Entity {
 		return this._entity.hasClass(Classes.activities.associateGrade.selected);
 	}
 
+	isDefault() {
+		if (!this._entity) {
+			return false;
+		}
+		return this._entity.hasClass(Classes.activities.associateGrade.default);
+	}
+
 	async selectScheme() {
-		if (!this.canChooseScheme()) {
+		if (!this.canChooseScheme() || this.isSelected()) {
```

Comment:
any chance each `GradeSchemeLinkedEntity` isn't updated so the entity may have the `selected` class, but not actually be the selected item in the UI?
Do we update the collection of grade schemes?
https://github.com/BrightspaceHypermediaComponents/activities/blob/v3.179.0/components/d2l-activity-editor/state/associate-grade.js#L45
Should that be called in `load`?
I also tried to optimize it so the categories and candidates aren't constantly re-fetched when the `associate-grade` entity is updated by doing the `this.gradeCategoryCollection || this.getGradeCategories();` thing. If schemes also did that, I think the collection might not update?

Reply:
I see your activities repo changes. They look good!